mlflow/models/evaluation/utils/trace.py

```diff
@@ -24,87 +35,118 @@ def configure_autologging_for_evaluation(enable_tracing: bool = True):
+        # because the evaluation code usually only use a subset of the supported flavors,
+        # hence we want to avoid unnecessary overhead of configuring all flavors.
+        @autologging_conf_lock
+        def _setup_autolog(module):
+            try:
+                autolog = get_autolog_function(flavor)
+
+                # If tracing is supported and not explicitly disabled, enable it.
+                if enable_tracing and _should_enable_tracing(flavor, global_config_snapshot):
                     new_config = {
                         k: False if k.startswith("log_") else v for k, v in original_config.items()
                     }
-
-                    # TODO: This check is a temporary fix for Databricks. This line explicitly
-                    # checks if the autolog is globally disabled via mlflow.autolog(disable=True).
-                    # In normal case, this is reflected to each flavor's configuration. However,
-                    # when user explicitly enables it for a specific flavor via
-                    # mlflow.<flavor>.autolog(), the global disablement does not take precedence,
-                    # which is intended.
-                    # However, in Databricks, sometimes mlflow.<flavor>.autolog() is automatically
-                    # called in the kernel startup, which is confused with the user's action. In
-                    # such cases, even when user disables autologging globally, the flavor-specific
-                    # autologging remains enabled. We are going to fix the Databricks side issue,
-                    # but until then we need to check the global configuration explicitly here.
-                    trace_disabled = global_config_snapshot.get("mlflow", {}).get("disable", False)
-
-                    new_config |= {"log_traces": not trace_disabled, "silent": True}
+                    new_config |= {"log_traces": True, "silent": True}
                     _kwargs_safe_invoke(autolog, new_config)
-                    trace_enabled_flavors.append(flavor)
-                elif flavor in AUTOLOGGING_INTEGRATIONS:
-                    # For flavors that do not support tracing, disable autologging
-                    autolog(disable=True)
-
-                flavor_to_original_config[flavor] = original_config
 
-        except Exception as e:
-            if isinstance(e, ImportError):
-                _logger.debug(
-                    f"Flavor {flavor} is not installed. Skip updating autologging. Error: {e!r}"
-                )
-            else:
-                _logger.info(
-                    f"Failed to update autologging configuration for flavor {flavor}. Error: {e!r}"
-                )
+                    global _IS_TRACE_MESSAGE_DISPLAYED
+                    if not _IS_TRACE_MESSAGE_DISPLAYED:
+                        _logger.info(
+                            "Tracing is temporarily enabled during the model evaluation "
```

Comment:
```suggestion
                            "Auto tracing is temporarily enabled during the model evaluation "
```
shouldn't we say `auto tracing`?